server/server.go

```diff
@@ -141,13 +141,15 @@ func (s *Server) ConnectionCount() int {
 func (s *Server) getToken() *Token {
 	start := time.Now()
 	tok := s.concurrentLimiter.Get()
+	metrics.TokenUsageGauge.Set(float64(s.concurrentLimiter.Allocated()) / float64(s.concurrentLimiter.Capacity()))
 	// Note that data smaller than one microsecond is ignored, because that case can be viewed as non-block.
 	metrics.GetTokenDurationHistogram.Observe(float64(time.Since(start).Nanoseconds() / 1e3))
 	return tok
 }
 
 func (s *Server) releaseToken(token *Token) {
 	s.concurrentLimiter.Put(token)
+	metrics.TokenUsageGauge.Set(float64(s.concurrentLimiter.Allocated()) / float64(s.concurrentLimiter.Capacity()))
```

Comment:
What about simply increase a counter in `Server::getToken` and release it in `Server::releaseToken`?

Reply:
Our goal is to calculate the usage of the tokens pool if we apply a gauge for `Token` we may need to export another metric that describes the `token-limit`.